### DISCUSSION TITLE
Features I Want in Future Updates

### DISCUSSION THREAD

**Discussion by DWrightAnimation** [keiwando/toonsquid#396] [General]
- Perspective and Warp Scaling Modes to be animatable like Freeform and Uniform 

- Animatable Effects (such as Blur(Gaussian and Motion), Glow, Color Curves, etc.) 

- More Advanced Rigging (like Mesh Warp, Inverse Kinematics, Underlying Skeleton for Rig, etc.) 

- Z-Axis Nudging (the only workaround for this is to have one layer of something animated in front of what you need to go in front and behind it, duplicate the layer, move the duplicate behind the base, and animate the opacity with Hold easing, it'd also make parallax a lot easier to do) 

- Pen Tool Advancements (such as controlling how thick and thin it gets throughout the line, being able to use other brushes for it instead of the same basic round brush, etc.) 

- Customizable Easing Curve (the presets are great, but it'd be nice to have the option to make your own timing, in addition to the presets) 

- Clearer Way to Manipulate the Animation Path for an object (arcs are the most important thing in animation, and it'd be nice if there was a way to manipulate those arcs in a more clear way) 

- Ability to Align Animation Drawings Scanned Into the iPad and Clean Them Up (OpenToonz has this, but it'd be nice if that feature could be on iPad) 

- Ability to Draw Outside of Canvas in Raster Mode (sometimes I need to sketch notes into the projects off camera, but shrinking the camera could lower the quality of my rough pencil tests. Vectors can draw outside of the canvas, but it doesn't feel the same as using the pencil brush, being able to make it as dark or light as I need) 

- Keyframe Library (Right now, when I use two poses that are far from each other in a symbol, and I make the timing anything but Hold, it'll go through the frames it took within the symbol to get to the pose, instead of making new inbetweens for the two poses) 

- Symbol Editing Advancements (Whenever I need to make a new mouth for a character, I can't see the character, or other symbols used for the character, to use as reference, and I don't know if it's possible to add a new drawing or pose to a symbol.

- Fixed Onion Skin Toggle: If we had this, it'd be a lot easier to make mouth poses akin to the Hanna-Barbera characters.

**Comment by VeiloPot** [comment]
I also had this idea where on effects, that if it can possible to add a stroke effect feature. Like for example, the text outline strokes and shape strokes, so I also thought what if we can add outlined strokes to symbols or rasterized stuff too?

**Comment by Matese19** [comment]
Features that I want the most:
- An adjustable grid that appears on the canvas and doesn’t move with the zooming in and out
- FlipaClip styled ruler tool
- Procreate Drawing assist tool
- Selection of draws on a single frame 
- Adjustable x axis
- Copy Paste for Transformation tool 
- Editing/Adding on to a symbol without having to reconstruct the whole thing
- Actual bones for character rigging 
- Easily being able to import a character rig into a project

**Comment by Pitchblack70** [comment]
I like what Dwright has to suggest about symbol editing advancements.

Currently, if you create a new drawing, there's no reference point. IF you could have a "light table" background of other layers to work from while editing... that would be FAB! (I was thinking of that possibility just today.)
